src/rgw/rgw_process.cc

```diff
@@ -319,5 +319,12 @@ int process_request(rgw::sal::RGWRadosStore* const store,
 	  << " ======"
 	  << dendl;
 
-  return (ret < 0 ? ret : s->err.ret);
+  if (ret < 0) {
+    if (s->expect_cont && s->content_length > 0) {
```

Comment:
i feel like this check needs to take into account whether we've actually replied to the client with 100 Continue - ie whether `dump_continue()` has been called. maybe `rgw_bucket_object_pre_exec()` should reset `expect_cont = false` after calling that?

Reply:
Actually this looks a  bit more trickier than that, the rfc mentions that the clients may send data even before waiting for a 100-continue response, our pre-exec code is executed after permission and param verification, so how would we deal with clients that wait for a ms and send the data anyway, for eg, boto does a wait for 1s before proceeding to send the data anyway 

https://github.com/boto/botocore/blob/develop/botocore/awsrequest.py#L158

(though boto is still broken for multi put, tests for this code were written using a simple asyncio aiohttp client which doesn't seem to be affected by httplib's consumption of 100-continue,  
https://gist.github.com/theanalyst/161d21767a78d2a9b4487c722aa6ea1a)

Reply:
rfc link https://tools.ietf.org/html/rfc7231#section-5.1.1